### PR TITLE
✨Bump cluster-api dependency to v1.4.1

### LINKED ISSUE #1832: Bump cluster-api to v1.4.1
/kind feature

**Describe the solution you'd like**
Start the process of updating the CAPI dependency in CAPV to the next minor release. Currently the RC version is out which should be used to start the consumption process and report any bugs.

### PR DESCRIPTION
**Pull request by srm09**
**What this PR does / why we need it**:
This patch bumps the CAPI module dependency to `v1.4.1` to integrate the latest minor release of CAPI with the CAPV release.

**Which issue(s) this PR fixes**:
Fixes #1832

**Special notes for your reviewer**:
n/a

**Release note**:
```release-note
Bump cluster-api dependency to v1.4.1
```

### CODE REVIEW

**Comment by srm09** [comment]
Blocked due to https://github.com/kubernetes-sigs/cluster-api/issues/8357
/hold

**Comment by srm09** [comment]
/retest pull-cluster-api-provider-vsphere-e2e 
The e2e test suite CI is failing due to the planned upgrades on the underlying infrastructure. Re-triggering.

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -18,7 +18,6 @@ spec:
       containers:
       - args:
         - --leader-elect
-        - --logtostderr
```
Comment (rikatz) [inline config/manager/manager.yaml:21]: can this be problematic? seems like something that changed on controller-gen maybe and just re-generated? not sure

**Comment by rikatz** [comment]
Overall lgtm, left a minor comment on removing logtostderr (can't remember if this was something from klog that does not exist anymore, or at least what is the motivation of changing this), but otherwise :shipit:  :)

File: config/manager/manager.yaml (hunk 1 above)
Comment (srm09) [reply config/manager/manager.yaml:21]: https://cluster-api.sigs.k8s.io/developer/providers/v1.3-to-v1.4.html#other mentions the removal of these deprecated flags from `component-base` which was necessary to get the CAPV manager running.

**Comment by srm09** [comment]
/test pull-cluster-api-provider-vsphere-e2e
Last failure was due to the older vCenter thumbprint value being used from the `ExternalSecret` CR defined for CAPV prow jobs. Retrying..

**Comment by srm09** [comment]
Adding back the `lgtm` label which was lost due to rebase.

**Pull request merged**